pkg/cmd/attestation/api/client.go

```diff
@@ -39,8 +61,7 @@ type httpClient interface {
 }
 
 type Client interface {
-	GetByRepoAndDigest(repo, digest string, limit int) ([]*Attestation, error)
```

Comment:
Instead of threading through the new predicateType parameter throughout the  nearly identical two methods, I collapsed the methods into one. The URL is chosen based on whether repo or owner is provided below.